src/ocm/components/AddHosts/day2Wizard/Day2ClusterDetails.tsx

```diff
@@ -1,47 +1,99 @@
+import { Stack, StackItem, FormGroup } from '@patternfly/react-core';
+import { Form, Formik, FormikHelpers, useFormikContext } from 'formik';
 import React from 'react';
-import { Stack, StackItem } from '@patternfly/react-core';
-import { Form, Formik, FormikHelpers } from 'formik';
 import {
   ClusterWizardStep,
   ClusterWizardStepHeader,
   CpuArchitecture,
   WizardFooter,
+  getFieldId,
+  RadioField,
+  Cluster,
+  LoadingState,
 } from '../../../../common';
+import { HostsNetworkConfigurationType, InfraEnvsService } from '../../../services';
 import { useModalDialogsContext } from '../../hosts/ModalDialogsContext';
 import { useDay2WizardContext } from './Day2WizardContext';
 import { Day2WizardNav } from './Day2WizardNav';
 import DiscoverImageCpuArchitectureControlGroup from '../../../../common/components/clusterConfiguration/DiscoveryImageCpuArchitectureControlGroup';
+import { handleApiError } from '../../../api';
+import { getDummyInfraEnvField } from '../../clusterConfiguration/staticIp/data/dummyData';
+import { InfraEnvsAPI } from '../../../services/apis';
 
 type Day2ClusterDetailValues = {
   cpuArchitecture: CpuArchitecture;
-  // hostsNetworkConfigurationType: 'dhcp' | 'static';
+  hostsNetworkConfigurationType: HostsNetworkConfigurationType;
+};
+
+const getDay2ClusterDetailInitialValues = async (clusterId: Cluster['id']) => {
+  try {
+    const { data: infraEnv } = await InfraEnvsService.getInfraEnv(clusterId, CpuArchitecture.x86);
+
+    return {
+      // TODO (multi-arch) improve to set the default value equal to the cpuArchitecture of Day1 cluster
+      cpuArchitecture: CpuArchitecture.x86,
+      hostsNetworkConfigurationType: infraEnv.staticNetworkConfig
+        ? HostsNetworkConfigurationType.STATIC
+        : HostsNetworkConfigurationType.DHCP,
+    };
+  } catch (error) {
+    console.error(error);
+  }
 };
 
 export const Day2ClusterDetails = () => {
   const { day2DiscoveryImageDialog } = useModalDialogsContext();
   const wizardContext = useDay2WizardContext();
+  const cluster = day2DiscoveryImageDialog.data.cluster;
   const { close } = day2DiscoveryImageDialog;
+  const [initialValues, setInitialValues] = React.useState<Day2ClusterDetailValues>();
+  const [isSubmitting, setSubmitting] = React.useState(false);
+
+  React.useEffect(() => {
+    const doItAsync = async () => {
+      const initialValues = await getDay2ClusterDetailInitialValues(cluster.id);
+      setInitialValues(initialValues);
+    };
+    void doItAsync();
+  }, [cluster.id]);
 
   const handleSubmit = React.useCallback(
-    (values: Day2ClusterDetailValues) => {
-      wizardContext.setSelectedCpuArchitecture(values.cpuArchitecture);
-      wizardContext.moveNext();
+    async (values: Day2ClusterDetailValues) => {
+      try {
+        setSubmitting(true);
+        const { data: infraEnvs } = await InfraEnvsAPI.list(cluster.id);
+        if (infraEnvs.every((infraEnv) => infraEnv.staticNetworkConfig)) {
```

Comment:
The code does that. In case the infraEnvs already use static IP, the UI sets `staticNetworkConfig` to `undefined` and the field isn't updated. In case the infraEnvs don't already use static IP, we send the dummy configuration. In case DHCP is chosen, we send `[]` which means "no static IP". In any case existing configurations won't be lost in case you navigate from the beginning of the form.

```
if (infraEnvs.every((infraEnv) => infraEnv.staticNetworkConfig)) {
          await InfraEnvsService.updateAll(cluster.id, {
            staticNetworkConfig:
              values.hostsNetworkConfigurationType === HostsNetworkConfigurationType.DHCP
                ? []
                : undefined,
          });
        } else {
          await InfraEnvsService.updateAll(cluster.id, {
            staticNetworkConfig:
              values.hostsNetworkConfigurationType === HostsNetworkConfigurationType.DHCP
                ? []
                : getDummyInfraEnvField(),
          });
        }
```

Reply:
I changed the code anyway. I hope this is clearer

```
const { data: infraEnvs } = await InfraEnvsAPI.list(cluster.id);
        const staticIPSet = infraEnvs.every((infraEnv) => infraEnv.staticNetworkConfig);

        const infraEnvUpdateParams: InfraEnvUpdateParams = {
          staticNetworkConfig:
            values.hostsNetworkConfigurationType === HostsNetworkConfigurationType.STATIC
              ? getDummyInfraEnvField()
              : [],
        };

        if (!(values.hostsNetworkConfigurationType === HostsNetworkConfigurationType.STATIC && staticIPSet)) {
          await InfraEnvsService.updateAll(cluster.id, infraEnvUpdateParams);
        }
```